### ISSUE TITLE
Kickstart-style migrations

### ISSUE THREAD

**Issue by atrauzzi** [FusionAuth/fusionauth-issues#560]
### Problem
Kickstarting is so helpful for the first-run setup, but I think what would be better is if we had a way to also provide configuration changes over time.

### Solution
Similar to how there are migration libraries for databases, the kickstart process should support running and tracking the "ran-state" of timestamped kickstart scripts, in order.

### Alternatives/workarounds
Right now I think the only way to make changes after a kickstart has happened is to do so manually through the UI, via direct SQL or via the API.

### Additional context
Given that kickstart already exists, this seems like a sensible addition.  Configuration upgrades would be possible by restarting the container with the additional files bound in.

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by robotdan** [comment]
Is this a dup of https://github.com/FusionAuth/fusionauth-issues/issues/526? or perhaps similar?

**Comment by atrauzzi** [comment]
Not quite.  The idea here would be support for "versioned" (timestamp is even sufficient) kickstart files to be run against the FustionAuth server after its initial deploy and to have those kickstart files originate from the orchestration layer.

So, think kickstart files that have mostly `PATCH` requests in them that have timestamps in their filenames.  Then FusionAuth ensures that all are run in order, once and only ever once.

**Comment by robotdan** [comment]
Got it, ok thanks for the clarification. Sort of like a record option to track all API requests and add then export the log as a kickstart file.

**Comment by atrauzzi** [comment]
Hmm, I'm not sure I follow on that description.  Maybe expand on it just a little more so I can be sure? :smile:

**Comment by mooreds** [comment]
I think that an alternative solution might be to create a Terraform provider for FusionAuth. I added an issue: https://github.com/FusionAuth/fusionauth-issues/issues/576

**Comment by atrauzzi** [comment]
That'd be super slick!

**Comment by kevcube** [comment]
I still don't feel the terraform provider is an adequate solution to this problem, is this something that will be worked on?
